frontend/packages/insights-plugin/src/components/InsightsPopup/index.tsx

```diff
@@ -0,0 +1,103 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { ChartDonut, ChartLegend, ChartLabel } from '@patternfly/react-charts';
+import { riskIcons, colorScale, legendColorScale, riskSorting, mapMetrics } from './mappers';
+import { PrometheusHealthPopupProps } from '@console/plugin-sdk';
+import './style.scss';
+
+const DataComponent: React.FC<DataComponentProps> = ({ x, y, datum }) => {
+  const Icon = riskIcons[datum.id];
+  return <Icon x={x} y={y - 5} fill={legendColorScale[datum.id]} />;
+};
+
+export const InsightsPopup: React.FC<PrometheusHealthPopupProps> = ({ responses, k8sResult }) => {
+  const resource = mapMetrics(responses[0].response);
+  const clusterId = _.get(k8sResult, 'data.spec.clusterID', '');
```

Comment:
nit: For new code, we prefer optional chaining. Console naming conventions would make this uppercase `ID`

```suggestion
  const clusterID = k8sResult?.data?.spec?.clusterID || '';
```